Add enrich policy GET API

hub-cap opened the pull request:
This commit wires up the Rest calls and Transport calls for GET enrich
policy, as well as tests and rest spec additions.

elasticmachine commented:
Pinging @elastic/es-core-features

martijnvg commented on x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/action/GetEnrichPolicyActionResponseTests.java:
```diff
@@ -0,0 +1,27 @@
+
+import static org.elasticsearch.xpack.enrich.EnrichPolicyTests.randomEnrichPolicy;
+
+public class GetEnrichPolicyActionResponseTests extends AbstractWireSerializingTestCase<GetEnrichPolicyAction.Response> {
```
I think this test should extend `AbstractSerializingTestCase`, so that xcontent serialization is also tested.

hub-cap commented:
I went ahead and removed this transport test as well, ala https://github.com/elastic/elasticsearch/pull/41383#discussion_r277076525, and fixed up the enrich store based on the review.

hub-cap commented:
@elasticmachine run elasticsearch-ci/1

martijnvg commented:
@elasticmachine run elasticsearch-ci/1

hub-cap commented:
@elasticmachine run elasticsearch-ci/1
